automation/test.sh

```diff
@@ -175,7 +175,9 @@ done
 set -e
 make bazel-build-images
 
+trap '{ echo "Dump kubevirt state:"; make dump; }' ERR
 make cluster-up
+trap - ERR
```

Comment:
> This disables the trap, right?

It does disable the trap.

> 
> It can happen that faulty deployment prevents kubevirt from getting deployed (with SR-IOV, we had accidents where the operator left all workers not schedulable). Can we extend the trap?

Can we extend the trap ? Yes - we can even 'not' disable it. 

The question is why would we do it; which operations are you concerned with ? The exact scenario you're pointing out is covered since the sriov operator is installed in `cluster-up` and we wait for all pods to be active: https://github.com/kubevirt/kubevirt/blob/master/cluster-up/cluster/kind-k8s-sriov-1.17.0/config_sriov.sh#L111

Please suggest a new location for disabling the trap if you think error-prone instructions are being left-out.

Reply:
I would keep the trap all the way to the point when another trap kicks in. So it would cover the whole installation of kubevirt and be stopped only once we start a test suite. Faulty deployment of SR-IOV can block deployment of KubeVirt. So I would disarm this on the second last line of this file. Tell me if I am too paranoid.

Reply:
Keep in mind we're already dumping the cluster state if pods don't  [reach Running state](https://github.com/kubevirt/kubevirt/blob/master/automation/test.sh#L225) or pod's containers don't [reach Ready state](https://github.com/kubevirt/kubevirt/blob/master/automation/test.sh#L240) within 5 minutes.

The trap I've added dumps cluster state on any sriov operator error, I think we're good.

Reply:
I see. all good then!